integration/bug-report-test.ts

```diff
@@ -59,27 +59,39 @@ test.beforeAll(async () => {
     ////////////////////////////////////////////////////////////////////////////
     files: {
       "app/routes/_index.tsx": js`
-        import { json } from "@remix-run/node";
-        import { useLoaderData, Link } from "@remix-run/react";
+        import { LoaderFunction, defer } from "@remix-run/node";
+        import { Await, useLoaderData } from "@remix-run/react";
+        import { Suspense } from "react";
 
-        export function loader() {
-          return json("pizza");
+        function sleep(ms: number) {
+          return new Promise((resolve) => setTimeout(resolve, ms));
         }
 
-        export default function Index() {
-          let data = useLoaderData();
+        export const loader: LoaderFunction = () => {
+          return defer({
+            immediate: "this data is available immediately",
+            deferred: sleep(3000).then(() => "this data is available after 3 seconds"),
+          });
+        };
+
+        export function Deferred() {
```

Comment:
This needs a `default` export to be treated as a UI route.  Without it, it's being treated as a [resource route](https://remix.run/docs/en/main/guides/resource-routes) which doesn't support `defer`

```suggestion
        export default function Deferred() {
```